src/libsync/propagateremotedeleteencrypted.h

```diff
@@ -29,13 +34,20 @@ class PropagateRemoteDeleteEncrypted : public QObject
     void slotTryLock(const QByteArray &folderId);
     void slotFolderLockedSuccessfully(const QByteArray &fileId, const QByteArray &token);
     void slotFolderEncryptedMetadataReceived(const QJsonDocument &json, int statusCode);
+    void slotdeleteNestedRemoteItemFinished();
+    void deleteNestedRemoteItemsFinished();
     void taskFailed();
+    void deleteNestedRemoteItem(const QString &filename);
 
     OwncloudPropagator *_propagator;
     SyncFileItemPtr _item;
     QByteArray _folderToken;
     QByteArray _folderId;
     bool _folderLocked = false;
+    QMap<QString, QString> _nestedItems; // Nested files and folders <encrypted_name, original_name>
```

Comment:
This is really more a mangled -> unmangled mapping. I'd rename this.

Reply:
@er-vin Well, if we speak about the fields that we put as key/value, then yes. But, when we speak about the list of items that we want to remove from the root folder we can't just say - **_mangledUnmangledItems** relating to a root folder? It'd be confusing. What I can do, is get rid of QMap, and create a structure MangledUnmangledMapping, and then use a **QVector<MangledUnmangled>**. The thing is, it doesn't really matter if we use a mapping from "Mangled to Unmangled", we can as well change the order to the opposite, but, we are still talking about nested files and folders in the root folder.
Which makes it even more confusing if we use **MangledtoUnmangled** in the name of a class member.